### PR TITLE
Add Register-FinOpsProvider

### LINKED ISSUE #248: [PowerShell] Register-FinOpsHubsProviders

### PR DESCRIPTION
**Pull request by saadmsft**
## 🛠️ Description
Added script for Register-FinOpsProvider functionality.

### Context : 
The Register-FinOpsHubProviders command registers the Azure resource providers required to deploy and operate a FinOps hub instance. To register a resource provider, you must have Contributor access (or the /register permission for each resource provider) for the entire subscription. Subscription readers can check the status of the resource providers but cannot register them. If you do not have access to register resource providers, please contact a subscription contributor or owner to run the Register-FinOpsHubProviders command.

Fixes #248 

## 🔬 How has this been tested?
- [x] 🫰 PS -WhatIf / az validate
- [x] 👍 Manually deployed + verified

### CODE REVIEW

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+<#
```
Comment (flanakin) [inline src/powershell/Public/Register-FinOpsHubProviders.ps1:3]: nit: Extra space
```suggestion

<#
```

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    .SYNOPSIS
+        Registers Azure resource providers required for FinOps hub.
```
Comment (flanakin) [inline src/powershell/Public/Register-FinOpsHubProviders.ps1:5]: Align the text with the label. Same for the others.
```suggestion
    .SYNOPSIS
    Registers Azure resource providers required for FinOps hub.
```

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        Shows what would happen if the command runs without actually running it.
+
+    .Description
+        The Register-FinOpsHubProviders command registers the Azure resource providers required to deploy 
+        and operate a FinOps hub instance. To register a resource provider, you must have 
+        Contributor access (or the /register permission for each resource provider) for the entire subscription. 
+        Subscription readers can check the status of the resource providers but cannot register them. 
+        If you do not have access to register resource providers, please contact a subscription contributor 
+        or owner to run the Register-FinOpsHubProviders command.
```
Comment (flanakin) [inline src/powershell/Public/Register-FinOpsHubProviders.ps1:21]: Should we keep the text on one line to make it easier for future changes to see what changed? That's what we do in markdown. When we break lines up, one added word can end up changing 20 lines, which makes it hard to understand what exactly changed.

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                }
+            }
+            # If the provider is already registered, logging a message saying so
+            else {
+                Write-Verbose -Message $($LocalizedData.ResourceProviderExists -f $provider)
+            }
+        }
+    }
+    catch [Microsoft.Azure.PowerShell.Cmdlets.ResourceManager.Runtime.AzurePowerShellException] {
+        Write-Verbose -Message $($LocalizedData.ErrorRegisteringProvider -f $_.Exception.Message)
```
Comment (flanakin) [inline src/powershell/Public/Register-FinOpsHubProviders.ps1:53]: Should we rethrow? I'm not familiar with PowerShell exception handling.

File: src/powershell/en-US/FinOpsToolkit.strings.psd1 [hunk 5]
```diff
@@ -10,7 +10,10 @@ ConvertFrom-StringData -StringData @'
     NewDirectory = Creating directory '{0}'.
     TemplateNotFound = Could not find template 'main.bicep' at path '{0}'.
     ContextNotFound = Could not retrieve Az context. Run Az-Login.
+    RegisterProvider =  Registering resource provider {0}.
+    ResourceProviderExists =  Resource provider {0} is already registered.
```
Comment (flanakin) [inline src/powershell/en-US/FinOpsToolkit.strings.psd1:14]: nit: `Registered` would probably be more accurate. (Don't forget to change the reference in the script.)
```suggestion
    ResourceProviderRegistered =  Resource provider {0} is already registered. 
```

**Review by flanakin — APPROVED** [review]

File: src/powershell/en-US/FinOpsToolkit.strings.psd1 (hunk 5 above)
Comment (saadmsft) [reply src/powershell/en-US/FinOpsToolkit.strings.psd1]: That's done, here and in script as well.

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 (hunk 1 above)
Comment (saadmsft) [reply src/powershell/Public/Register-FinOpsHubProviders.ps1:4]: Done

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 (hunk 2 above)
Comment (saadmsft) [reply src/powershell/Public/Register-FinOpsHubProviders.ps1]: made changes requested

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 (hunk 3 above)
Comment (saadmsft) [reply src/powershell/Public/Register-FinOpsHubProviders.ps1]: You are right, I put that in one line.

File: src/powershell/Public/Register-FinOpsHubProviders.ps1 (hunk 4 above)
Comment (saadmsft) [reply src/powershell/Public/Register-FinOpsHubProviders.ps1:49]: Yes, Correct re-throwing it with the custom error message. Saved the string in psd1.

**Review by flanakin — APPROVED** [review]

**Pull request merged**